contracts/vaults/base/RibbonVault.sol

```diff
@@ -0,0 +1,633 @@
+    address public immutable MARGIN_POOL;
+
+    // GNOSIS_EASY_AUCTION is Gnosis protocol's contract for initiating auctions and placing bids
+    // https://github.com/gnosis/ido-contracts/blob/main/contracts/EasyAuction.sol
+    address public immutable GNOSIS_EASY_AUCTION;
+
+    /************************************************
+     *  EVENTS
+     ***********************************************/
+
+    event Deposit(address indexed account, uint256 amount, uint16 round);
+
+    event InitiateWithdraw(address account, uint256 shares, uint16 round);
+
+    event Withdraw(address indexed account, uint256 amount, uint256 share);
+
+    event InstantWithdraw(
+        address indexed account,
+        uint256 amount,
+        uint16 round
+    );
+
+    event Redeem(address indexed account, uint256 share, uint16 round);
+
+    event WithdrawalFeeSet(uint256 oldFee, uint256 newFee);
+
+    event ManagementFeeSet(uint256 managementFee, uint256 newManagementFee);
+
+    event PerformanceFeeSet(uint256 performanceFee, uint256 newPerformanceFee);
+
+    event CapSet(uint256 oldCap, uint256 newCap, address manager);
+
+    event CollectVaultFees(
+        uint256 performanceFee,
+        uint256 vaultFee,
+        uint256 round
+    );
+
+    /************************************************
+     *  CONSTRUCTOR & INITIALIZATION
+     ***********************************************/
+
+    /**
+     * @notice Initializes the contract with immutable variables
+     * @param _weth is the Wrapped Ether contract
+     * @param _usdc is the USDC contract
+     * It's important to bake the _factory variable into the contract with the constructor
+     * If we do it in the `initialize` function, users get to set the factory variable and
+     * subsequently the adapter, which allows them to make a delegatecall, then selfdestruct the contract.
+     */
+    constructor(
+        address _weth,
+        address _usdc,
+        address _gammaController,
+        address _marginPool,
+        address _gnosisEasyAuction
+    ) {
+        require(_weth != address(0), "!_weth");
+        require(_usdc != address(0), "!_usdc");
+        require(_gnosisEasyAuction != address(0), "!_gnosisEasyAuction");
+        require(_gammaController != address(0), "!_gammaController");
+        require(_marginPool != address(0), "!_marginPool");
+
+        WETH = _weth;
+        USDC = _usdc;
+        GAMMA_CONTROLLER = _gammaController;
+        MARGIN_POOL = _marginPool;
+        GNOSIS_EASY_AUCTION = _gnosisEasyAuction;
+    }
+
+    /**
+     * @notice Initializes the OptionVault contract with storage variables.
+     */
+    function baseInitialize(
+        address _owner,
+        address _feeRecipient,
+        uint256 _managementFee,
+        uint256 _performanceFee,
+        string memory tokenName,
+        string memory tokenSymbol,
+        Vault.VaultParams calldata _vaultParams
+    ) internal initializer {
```

Comment:
Need to check on this but this will probably throw because you have two `initializer` modifiers that are chained together

Reply:
theta vault tests passed, which includes [initialization](https://github.com/ribbon-finance/ribbon-v2/blob/bb0f9121938112aaed0b51ff5e679015f62406c1/test/RibbonThetaVault.ts#L417)